modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.models.service.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.models.common.exceptions.VariableInvalidException;
+import org.eclipse.xpanse.modules.models.servicetemplate.DeployVariable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * The class is used to validate deployment variables.
+ */
+@Slf4j
+@Component
+public class ServiceVariablesJsonSchemaValidator {
+
+    private static final ObjectMapper jsonMapper = new ObjectMapper();
+
+    /**
+     * Check validation of deploy property map by list of deployVariables in registered service.
+     *
+     * @param deployVariables list of deployVariables in registered service
+     * @param deployProperty  deploy property map
+     */
+    public void validateDeployVariables(List<DeployVariable> deployVariables,
+                                        Map<String, String> deployProperty,
+                                        String jsonObjectSchemaStr) {
+        if (!CollectionUtils.isEmpty(deployVariables) && !CollectionUtils.isEmpty(deployProperty)) {
+            try {
+                Locale.setDefault(Locale.ENGLISH);
```

Comment:
this wil overwrite for the entire JVM. Instead use the 'SchemaValidatorsConfig' option in the link which i gave you.